src/containers/BooksForm.js

```diff
@@ -1,18 +1,58 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions/index';
 
 const bookCategories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
 
-const BooksForm = () => (
-  <form>
-    <label htmlFor="title-input">Title</label>
-    <input id="title-input" type="text" />
-    <select>
-      {bookCategories.map((category, index) => (
-        <option value="category" key={index}>{category}</option>
-      ))}
-    </select>
-    <input type="submit" />
-  </form>
-);
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: bookCategories[0],
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
 
-export default BooksForm;
+  handleChange(event) {
+    this.setState({ [event.target.id]: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const newBook = this.state;
+    const { createBook } = this.props;
+    createBook(newBook);
+    this.setState({ title: '', category: bookCategories[0] });
+    document.getElementById(bookCategories[0]).selected = true;
+  }
+
+  render() {
+    const { title } = this.state;
+    return (
+      <form>
+        <label htmlFor="title">Title</label>
+        <input
+          id="title"
+          type="text"
+          name="title"
+          value={title}
+          onChange={this.handleChange}
+        />
+        <select id="category" onChange={this.handleChange}>
```

Comment:
Let's make the select element controlled as well. Add `value={category}` prop to it to update based on the state.